crates/rust-analyzer/src/reload.rs

```diff
@@ -305,8 +305,42 @@ impl GlobalState {
 
         if self.proc_macro_clients.is_empty() {
             if let Some((path, args)) = self.config.proc_macro_srv() {
-                self.proc_macro_clients = (0..self.workspaces.len())
-                    .map(|_| {
+                self.proc_macro_clients = self
+                    .workspaces
+                    .iter()
+                    .map(|ws| {
+                        let mut args = args.clone();
+                        let mut path = path.clone();
+
+                        if let ProjectWorkspace::Cargo { sysroot, .. } = ws {
```

Comment:
So at this point in time, we don't have any support for `ProjectWorkspace::Json`, even if `sysroot` is set?

Reply:
Did proc-macros every work for project.json workspaces actually? From the looks of it we never even build build scripts for them since we can't really do that

Reply:
Yes, proc macros should work with project.json workspaces. We have a field that allows specifying the path of the compiled proc macro.

Reply:
Can you point me to where the proc-macro-srv is spawned for ProjectWorkspace::Json workspaces?

Reply:
https://github.com/rust-lang/rust-analyzer/blob/b1e9bcddc2dd83568ea5a5ed15a51ef2d1cdd89f/crates/project-model/src/workspace.rs#L484 I think.

Reply:
@woody77 that match is definitely missing an arm I forgot to add when I added the additional field to rust-project.json's schema. I also didn't have any projects to test it with. I encourage you to open the small PR to fix it!